### PR TITLE
Map getLocation and SetBounds

### PR DESCRIPTION
**Pull request by Mrunzzz**

### CODE REVIEW

File: app/src/main/java/in/ac/iitb/gymkhana/iitbapp/MainActivity.java [hunk 1]
```diff
@@ -47,6 +52,15 @@ protected void onCreate(Bundle savedInstanceState) {
 
         NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
         navigationView.setNavigationItemSelectedListener(this);
+        if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION) !=
```
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/iitbapp/MainActivity.java]: Permission request should be part on the `MapFragment` and not `MainActivity`

File: app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java [hunk 2]
```diff
@@ -34,6 +37,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+            googleMap.setMyLocationEnabled(true);
+            googleMap.getUiSettings().setMyLocationButtonEnabled(true);
+
+        }
+
+
+        LatLngBounds iitbBounds=new LatLngBounds(new LatLng(19.1249000, 72.9046000),new LatLng(19.143522, 72.920000));
+        googleMap.setLatLngBoundsForCameraTarget(iitbBounds);
+        googleMap.setMaxZoomPreference(30);
+        googleMap.setMinZoomPreference(18);
```
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java]: I don't think we need zoom bounds. The map feels very non intuitive as of now.

File: app/src/main/java/in/ac/iitb/gymkhana/iitbapp/MainActivity.java (hunk 1 above)
Comment (Mrunzzz) [inline app/src/main/java/in/ac/iitb/gymkhana/iitbapp/MainActivity.java]: Then After asking for permission the button doesnt appear unless u reopen the fragment.. How can that be done?

File: app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java (hunk 2 above)
Comment (Mrunzzz) [inline app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java]: If u remove min zoom bound, the map wont get restricted to the iitb area... Just that u cannot zoom in outside the area in that case

File: app/src/main/java/in/ac/iitb/gymkhana/iitbapp/MainActivity.java (hunk 1 above)
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/iitbapp/MainActivity.java]: You need to override `onPermissionsResult()`. Check the code in [Tum Tum Tracker](https://github.com/unstablebrainiac/Tum-Tum-Tracker/) to get an idea.

File: app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java (hunk 2 above)
Comment (unstablebrainiac) [inline app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java]: Download the [Google I/O 2017 app](https://play.google.com/store/apps/details?id=com.google.samples.apps.iosched&hl=en) to know **what** to do. Google hasn't made the source code public yet so you'll have to figure out **how** to do it yourself.

Comment (Mrunzzz) [inline app/src/main/java/in/ac/iitb/gymkhana/iitbapp/fragments/MapFragment.java]: ok sure

**Comment by unstablebrainiac** [comment]
Nice work, sorry for the delay

**Pull request merged**